demos/gesture_recognition_demo/cpp_gapi/main.cpp

```diff
@@ -0,0 +1,210 @@
+                                                                                    batch_constant_FPS)));
+
+        std::string mainWindowName = "Gesture Recognition demo G-API";
+        std::string gestureWindowName = "Gesture";
+
+        cv::Size graphSize{static_cast<int>(frame_size.width / 4), 60};
+        Presenter presenter(FLAGS_u, frame_size.height - graphSize.height - 10, graphSize);
+
+        /** Save output result **/
+        cv::VideoWriter videoWriter;
+        if (!FLAGS_o.empty() && !videoWriter.open(FLAGS_o, cv::VideoWriter::fourcc('M', 'J', 'P', 'G'),
+                                                  cap->fps(), frame_size)) {
+            throw std::runtime_error("Can't open video writer");
+        }
+
+        /** Fill labels container from file with classes **/
+        const auto labels = fill_labels(FLAGS_c);
+        size_t current_id = 0;
+        size_t id = 0;
+        int gesture = 0;
+
+        /** Configure drawing utilities **/
+        Visualizer visualizer(FLAGS_no_show, mainWindowName, gestureWindowName, labels, FLAGS_s);
+
+        bool isStart = true;
+        const auto startTime = std::chrono::steady_clock::now();
+        pipeline.start();
+        while (pipeline.pull(std::move(out_vector))) {
+            /** Put FPS to frame**/
+            if (isStart) {
+                metrics.update(startTime, out_frame, { 10, 22 }, cv::FONT_HERSHEY_COMPLEX,
+                               0.65, { 200, 10, 10 }, 2, PerformanceMetrics::MetricTypes::FPS);
+                isStart = false;
+            } else {
+                metrics.update({}, out_frame, { 10, 22 }, cv::FONT_HERSHEY_COMPLEX,
+                               0.65, { 200, 10, 10 }, 2, PerformanceMetrics::MetricTypes::FPS);
+            }
+
+            /** Display system parameters **/
+            presenter.drawGraphs(out_frame);
+            /** Display the results **/
+            visualizer.show(out_frame, out_detections, out_label_number, current_id, gesture);
+            gesture = 0;
+
+            if (videoWriter.isOpened()) {
+                videoWriter.write(out_frame);
+            }
+
+            /** Controls **/
+            int key = cv::waitKey(1);
+            if      (key == 0x1B) break;  // (esc button) exit
+            else if (key >= 48 && key <= 57) id = key - 48; // buttons for person id
+            else if (key == 0x0D) out_label_number = -1; // (Enter) reset last gesture
+            else if (key == 'f') gesture = 1; // next gesture
+            else if (key == 'b') gesture = -1; // prev gesture
+            else
+                presenter.handleKey(key);
+
+            /** Share id with graph **/
+            if (id < out_detections.size()) {
+                current_person_id = id;
+                current_id = id;
+            }
+        }
+        /** Destroy windows if exist **/
+        visualizer.finalize();
```

Comment:
You don't need this. OpenCV will destroy windows on its own.
I think I missed similar line in interactive_face_detection_demo

Reply:
Removed. Thanks.